src/modules/core/containers/user-header-navigation/user-header-navigation.tsx

```diff
@@ -21,23 +23,47 @@ const userLinks: { href: string; title: string }[] = [
 ];
 
 export const UserHeaderNavigation: FC<UserHeaderNavigationProps> = () => {
+  const session = useSession();
   const pathname = usePathname();
 
+  if (session.status === 'unauthenticated') {
+    return null;
+  }
+
+  if (session.status === 'authenticated') {
```

Comment:
On a side note I think we should allow user make logout somehow even if they didn't completed onboarding.